obspy/io/stationxml/core.py

```diff
@@ -751,25 +775,20 @@ def _write_stationxml(inventory, file_or_file_object, validate=False,
         break
 
     if data_availability:
-        root = etree.Element(
-            "FDSNStationXML",
-            attrib={
-                ("{http://www.w3.org/2001/XMLSchema-instance}"
-                 "schemaLocation"): "http://www.fdsn.org/xml/station/1 "
-                "http://www.fdsn.org/xml/station/fdsn-station+"
-                "availability-1.0.xsd",
-                "schemaVersion": SCHEMA_VERSION},
-            nsmap={None: "http://www.fdsn.org/xml/station/1",
-                   "xsi": "http://www.w3.org/2001/XMLSchema-instance"}
-        )
-    else:
-        root = etree.Element(
-            "FDSNStationXML",
-            attrib={
-                "xmlns": "http://www.fdsn.org/xml/station/1",
-                "schemaVersion": SCHEMA_VERSION}
-        )
-
+        attrib["{http://www.w3.org/2001/XMLSchema-instance}"
+               "schemaLocation"] = (
+            "http://www.fdsn.org/xml/station/1 "
+            "http://www.fdsn.org/xml/station/fdsn-station+"
+            "availability-1.0.xsd")
+        if "xsi" in nsmap:
+            msg = ("Custom namespace mappings do not allow redefinition of "
+                   "StationXML availability namespace (key `xsi`). Use other "
+                   "namespace abbreviations for custom namespace tags.")
+            raise ValueError(msg)
```

Comment:
I added a new test titled _test_write_with_extra_tags_namespace_redef_ too confirm that these exceptions are raised when appropriate.